### PR TITLE
Create low level Client.SendMail utility function

### PR DESCRIPTION
**Pull request by petersanchez**
For cases when more control is needed than what the helper SendMail provides.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,7 +45,41 @@ func main() {
...
+
+func main() {
+	// Setup connection to mail server, return Client instance
+	c, err := smtp.Dial("mail.example.com:25")
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	// Connect to the server, authenticate, set the sender and recipient,
```
Comment (emersion) [inline README.md:70]: Remove "authenticate" here

File: client.go [hunk 2]
```diff
@@ -502,6 +502,33 @@ func (c *Client) LMTPData(statusCb func(rcpt string, status *SMTPError)) (io.Wri
 	return &dataCloser{c, c.Text.DotWriter(), statusCb}, nil
 }
 
+// SendMail is the low level email delivery agent.
```
Comment (emersion) [inline client.go:505]: It would be nice to describe exactly what this function does, and what it doesn't do. Can take inspiration from the existing `SendMail` function. e.g. something like:

```go
// SendMail sends an email from address from, to addresses to, with message r.
//
// This function does not start TLS, nor does it perform authentication. Use
// StartTLS and Auth before-hand if desirable.
//
// The addresses in the to parameter are the SMTP RCPT addresses.
//
// The r parameter should be an RFC 822-style email with headers first, a blank
// line, and then the message body. The lines of r should be CRLF terminated.
```

File: client.go [hunk 3]
```diff
@@ -456,16 +456,16 @@ func (d *dataCloser) Close() error {
...
+
+	_, _, err := d.c.Text.ReadResponse(250)
+	if err != nil {
+		if protoErr, ok := err.(*textproto.Error); ok {
+			return toSMTPErr(protoErr)
 		}
-		return nil
+		return err
 	}
+	return nil
```
Comment (emersion) [inline client.go:468]: Hm, why this change in `dataCloser.Close`? I don't think it's correct, I'd assume it breaks LMTP.

**Comment by emersion** [comment]
When the review comments are addressed, please squash all of the commits together into a single logical commit.

File: client.go (hunk 3 above)
Comment (petersanchez) [reply client.go:468]: Ah actually I meant to ask you about this. gopls was complaining about the block so I adjusted it and ran tests (passed) so I assumed it was fine. Was going to ask first and forgot. I'll revert this though.

Comment (emersion) [reply client.go:468]: Oh, I missed the `return` in the `if` branch above. So I guess both are the same.

I'd prefer to keep it the way it was before though, because it makes it clearer what the two cases are.

**Comment by petersanchez** [comment]
@emersion All should be resolved now. Let me know if you'd like any other changes to this.

**Review by emersion — APPROVED** [review]
LGTM, thanks!
